src/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs

```diff
@@ -514,11 +502,18 @@ internal static char ParseFormatSpecifier(string format, out int digits)
             return (char)0; // Custom format
         }
 
-        private static string FormatBigIntegerToHexString(BigInteger value, char format, int digits, NumberFormatInfo info)
+        private static string FormatBigIntegerToHex(BigInteger value, char format, int digits, NumberFormatInfo info)
         {
-            StringBuilder sb = new StringBuilder();
-            byte[] bits = value.ToByteArray();
-            string fmt = null;
+            Debug.Assert(format == 'x' || format == 'X');
+
+            // Get the bytes that make up the BigInteger.
+            Span<byte> bits = stackalloc byte[64]; // arbitrary limit to switch from stack to heap
```

Comment:
can we call this bytes, not bits? I was confused at first.

Reply:
This is the same name that was there; I've just changed its type from `byte[]` to `Span<byte>`.  Someone more familiar with BigInteger can choose to rename this as part of a subsequent cleanup if desired, but this `bits` terminology is used in other places.